applications/GeoMechanicsApplication/python_scripts/geomechanics_U_Pw_solver.py

```diff
@@ -0,0 +1,459 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+from KratosMultiphysics.python_solver import PythonSolver
+
+# Check that applications were imported in the main script
+# DEPRECATION: "CheckRegisteredApplications" is not needed any more and can be safely removed
+#KratosMultiphysics.CheckRegisteredApplications("GeoMechanicsApplication")
+
+# Import applications
+import KratosMultiphysics.GeoMechanicsApplication as KratosGeo
+import KratosMultiphysics.StructuralMechanicsApplication as KratosStructure
+
+# Import base class file
+import KratosMultiphysics.GeoMechanicsApplication.geomechanics_solver as GeoSolver
+
+def CreateSolver(model, custom_settings):
+    return UPwSolver(model, custom_settings)
+
+class UPwSolver(GeoSolver.GeoMechanicalSolver):
+    '''Solver for the solution of displacement-pore pressure coupled problems.'''
+
+    def __init__(self, model, custom_settings):
+        super(UPwSolver,self).__init__(model, custom_settings)
+
+        # There is only a single rank in OpenMP, we always print
+        self._is_printing_rank = True
+
+
+        KratosMultiphysics.Logger.PrintInfo("UPwSolver", "Construction of UPwSolver finished.")
+
+    @classmethod
+    def GetDefaultParameters(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "solver_type": "geomechanics_U_Pw_solver",
+            "model_part_name": "PorousDomain",
+            "domain_size": 2,
+            "start_time": 0.0,
+            "model_import_settings":{
+                "input_type": "mdpa",
+                "input_filename": "unknown_name"
+            },
+            "material_import_settings" :{
+                "materials_filename": ""
+            },
+            "time_stepping": {
+                "end_time" : 1.0,
+                "time_step": 0.1
+            },
+            "buffer_size": 2,
+            "echo_level": 0,
+            "reform_dofs_at_each_step": false,
+            "clear_storage": false,
+            "compute_reactions": false,
+            "move_mesh_flag": false,
+            "nodal_smoothing": false,
+            "reset_displacements":  false,
+            "solution_type": "quasi_static",
+            "scheme_type": "Newmark",
+            "newmark_beta": 0.25,
+            "newmark_gamma": 0.5,
+            "newmark_theta": 0.5,
+            "rayleigh_m": 0.0,
+            "rayleigh_k": 0.0,
+            "strategy_type": "newton_raphson",
+            "convergence_criterion": "Displacement_criterion",
```

Comment:
```suggestion
            "convergence_criterion": "displacement_criterion",
```
Otherwise the convergence criteria factory would throw an error if not provided.

Reply:
OK. Now I see the function below. It works. Feel free to dismiss this comment.